src/core_modules/capture-core/components/Pages/MainPage/EventWorkingLists/DataSourceSetup/EventWorkingListsDataSourceSetup.component.js

```diff
@@ -55,10 +56,17 @@ export const EventWorkingListsDataSourceSetup = ({
         columns,
     ]);
 
+    const dataSourceArray = useMemo(() =>
+        recordsOrder && recordsOrder
+            .map(id => dataSource[id]), [
+        dataSource,
+        recordsOrder,
+    ]);
```

Comment:
Dont know what is your opinion here but I find that there will be more consistent if we returned an empty array in case records order is not defined. 

This way we know that this is _always_ an array but it can sometimes just be empty.

Reply:
For me, it depends. Most of the time I agree with you, here it doesn't really matter, because the value should be defined after the data is fetched from the api.

Reply:
Also sometimes you will utilize this distinction, like I do in the offline lists in an upcoming PR.

Reply:
Yep okay that makes sense to me as well